lib/_stream_writable.js

```diff
@@ -362,27 +388,34 @@ function onwriteDrain(stream, state) {
 function clearBuffer(stream, state) {
   state.bufferProcessing = true;
   var entry = state.bufferedRequest;
+  var bufferedRequests = state.bufferedRequests;
+
+  state.bufferedRequests = 0;
 
   if (stream._writev && entry && entry.next) {
     // Fast case, write everything using _writev()
-    var buffer = [];
-    var cbs = [];
+    var buffer = new Array(bufferedRequests);
```

Comment:
@thefourtheye Splitting the buffers and callbacks into two separate arrays is necessary to be able to pass all of the buffers to `_writev` separate from their callbacks. In many cases the list of buffers is forwarded to the C++ layer which expects an array of buffers.